.github/workflows/validate_internal_docs_links.yml

```diff
@@ -0,0 +1,23 @@
+name: Validate internal documentation links
+
+"on": [pull_request]
+
+jobs:
+    tests:
+        name: mkdocs serve strict
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v4.0.0
+
+            - name: Set up Python
+              uses: actions/setup-python@v4.7.0
+              with:
+                  python-version: 3.8
+
+            -   name: Install deps
+                run: |
+                    pip install --upgrade --upgrade-strategy eager -r requirements-dev.txt -e .
+                    pip check
+
+            - name: launch mkdocs in strict mode
+              run: timeout 20 mkdocs serve --dev-addr 127.0.0.1:8000 --strict || code=$?; if [[ $code -ne 124 && $code -ne 0 ]]; then exit $code; fi
```

Comment:
Nice! Isn't the `serve` and wait redundant? Like, if you just do `mkdocs build -s`, wouldn't it croak with any link issue?

Reply:
I vaguely remember that using mkdocs build is where I started from in my Atlas PR, but it didn't work, I don't remember why. If you use the serve, then the timeout is necessary to set a stop at the check, because the action otherwise never ends (because it's a server running). But I'll try again with the build instead.

Reply:
Yes, it was this one, basically there is always this warning and it makes the action end with error:

<img width="995" alt="image" src="https://github.com/Clinical-Genomics/scout/assets/28093618/a6f75950-b26d-4795-b59b-2b606a26b39d">

Reply:
Mmm but wait, after I break a link the action doesn't fail. I need to fix this

Reply:
Ah no, there was a release (1.5) this summer where they changed the behavior of the links validation! --> https://github.com/mkdocs/mkdocs/releases

Reply:
Ok this is the behavior with a broken link and new validation settings in mkdocs.yml:

mkdocs build never really gets to the link validation step, because it fails before:

<img width="1146" alt="image" src="https://github.com/Clinical-Genomics/scout/assets/28093618/dbf63969-6a07-4099-8ab0-00a7d38dd85a">

mkdocs serve works: 🆗 

<img width="1169" alt="image" src="https://github.com/Clinical-Genomics/scout/assets/28093618/45b39dd1-0a22-4f69-a094-df0c9d63b9e8">

Reply:
We talked about it, but to document, I think this can be fixed with using a localhost dev_addr, just like you do in the validation with serve. It should be transparent to local users, and not affect the final site.